File: repeated_string.js
"use strict"

const fs = require("fs")

process.stdin.resume()
process.stdin.setEncoding("utf-8")

let inputString = ""
let currentLine = 0

process.stdin.on("data", (inputStdin) => {
  inputString += inputStdin
})

process.stdin.on("end", (_) => {
  inputString = inputString
    .replace(/\s*$/, "")
    .split("\n")
    .map((str) => str.replace(/\s*$/, ""))

  main()
})

function readLine() {
  return inputString[currentLine++]
}

// Complete the repeatedString function below.
function repeatedString(s, n) {
  let a1 = 0,
    a2 = 0
  let mod = n % s.length

  for (let i = 0; i < s.length; i++) {
    if (s[i] == "a") {
      a1++
      if (i < mod) a2++
    }
  }

  return ((n - mod) / s.length) * a1 + a2
}

function main() {
  const ws = fs.createWriteStream(process.env.OUTPUT_PATH)

  const s = readLine()

  const n = parseInt(readLine(), 10)

  let result = repeatedString(s, n)

  ws.write(result + "\n")

  ws.end()
}

File: repeated_string.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the repeatedString function below.
def repeatedString(s, n):
    a1, a2 = 0, 0
    mod = n % len(s)

    for i in range(len(s)):
        if s[i] == 'a':
            a1 += 1
            if i < mod:
                a2 += 1

    return int((n - (mod)) / len(s)) * a1 + a2

    # Another option that works with small n:
    # while s_len < n:
    #     for i in s:
    #         if i == 'a':
    #             a += 1
    #         s_len += 1
    #         if s_len == n:
    #             break
    # return a


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    n = int(input())

    result = repeatedString(s, n)

    fptr.write(str(result) + '\n')

    fptr.close()
